### PR TITLE
feat(protocol-designer): Add set temperature form

### LINKED ISSUE #4307: PD: Temp Deck - Set temp step in timeline
# User Story
As a user, I should be able to add a step involving the temperature module to my protocol timeline.

# UI Design 
New step list: https://zpl.io/aR9ljQN
Temp step in timeline: https://zpl.io/adYlqE7

# Acceptance Criteria
Row 1 should contain the following info:
- [ ] The name of the labware on the module
- [ ] The status of the module that step is creating (ie, "Go 4C")

### PR DESCRIPTION
**Pull request by Kadee80**
## overview

This PR closes #4307 by adding the set temperature form (for temperature modules). It also refactors the labware dropdown field to separate the container from a now reusable `StepFormDropdown` which is used in the `TemperatureForm` to select the labware associated with the temperature module. 

Screen Shot 2019-12-04 at 9 39 17 AM

## changelog

- refactor: Split labware dropdown container and presentational component 
- feat: Add module ui selectors to get dropdown options for labware associated with different modules on the deck
- feat(protocol-designer): Add set temperature form

## review requests

- Make a protocol with a temperature module
- Add a labware to the temperature module (maybe a nickname to said labware)
- Add a temperature step

- [ ] Labware on temperature module appears in dropdown (full name if no nickname, nickname shows if present)
- [ ] Labware options value is the module ID - not the labware ID
- [ ] Selecting change temperature conditionally renders temperature text field
- [ ] Selecting deactivate... hides the temperature text field

NOTE:
Step generation button, step label in timeline, and step form title are all currently TEMPERATURE, this differs from the latest designs but follows the existing patterns for forms. Will update after a discussion with design in a follow up PR - along with what we decide to do with the dropdown if there is no labware on the temperature module.

### CODE REVIEW

File: protocol-designer/src/components/StepEditForm/fields/StepFormDropdown.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    onFieldFocus,
+  } = props
+  return (
+    // TODO: BC abstract e.currentTarget.value inside onChange with fn like onChangeValue of type (value: mixed) => {}
+    <StepField
+      name={name}
+      focusedField={focusedField}
+      dirtyFields={dirtyFields}
+      render={({ value, updateValue, errorToShow }) => {
+        // blank out the dropdown if labware id does not exist
```
Comment (IanLondon) [inline protocol-designer/src/components/StepEditForm/fields/StepFormDropdown.js]: "blank out options where values are falsey"?

File: protocol-designer/src/ui/modules/selectors.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import { type ModuleType } from '@opentrons/shared-data'
+import type { Selector } from '../../types'
+import type {
+  ModuleOnDeck,
+  LabwareOnDeck,
+  ModuleEntity,
+  InitialDeckSetup,
+} from '../../step-forms/types'
+
+export function getModuleOnDeckByType(
```
Comment (IanLondon) [inline protocol-designer/src/ui/modules/selectors.js]: to make it more clear they're not selectors, maybe I'd underscore these non-selector fns eg `_getModuleOnDeckByType` or if we wanna go crazy, move them to a new `./utils.js`?

File: protocol-designer/src/ui/modules/selectors.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  initialDeckSetup: InitialDeckSetup,
+  type: ModuleType
+): ?ModuleEntity {
```
Comment (IanLondon) [inline protocol-designer/src/ui/modules/selectors.js]: would this be `?ModuleOnDeck`? Going by L22

File: protocol-designer/src/ui/modules/selectors.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+          {
+            name: `${prefix} ${nicknamesById[labware.id]}`,
+            value: module.id,
+          },
+        ]
+      : []
+  return options
+}
+
+/** Returns drowdown option for labware placed on magnetic module */
```
Comment (IanLondon) [inline protocol-designer/src/ui/modules/selectors.js]: `drowdown`

File: protocol-designer/src/ui/modules/selectors.js (hunk 2 above)
Comment (Kadee80) [reply protocol-designer/src/ui/modules/selectors.js]: I like moving them to utils.js! Will do so now

File: protocol-designer/src/ui/modules/selectors.js (hunk 3 above)
Comment (Kadee80) [reply protocol-designer/src/ui/modules/selectors.js]: 👍

File: protocol-designer/src/components/StepEditForm/fields/StepFormDropdown.js (hunk 1 above)
Comment (Kadee80) [reply protocol-designer/src/components/StepEditForm/fields/StepFormDropdown.js]: Getting wireframes for this soon, and we can update accordingly in the follow up PR and also address the other things listed in notes!

**Pull request merged**